assets/src/demo/app.js

```diff
@@ -0,0 +1,188 @@
+import { withSelect, withDispatch } from '@wordpress/data';
+import { isModelEntityOfModel } from '@eventespresso/validators';
+import './app.css';
+
+class EventInput extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = { eventId: props.eventId || 0 };
+	}
+
+	setEventId = ( id ) => {
+		this.setState( { eventId: id } );
+	};
+
+	onEventSubmit = () => this.props.onEventClick( this.state.eventId );
+
+	render() {
+		return <div>
+			<TextControl
+				label={ 'Enter the EventId here' }
+				value={ this.state.eventId }
+				onChange={ this.setEventId }
+			/>
+			<Button
+				label={ 'Retrieve Event' }
+				isPrimary={ true }
+				onClick={ this.onEventSubmit }
+			>
+				Retrieve Event
+			</Button>
+		</div>;
+	}
+}
+
+const DatesList = ( { datetimes, onDateClone } ) => {
+	function getDatetimesList() {
+		if ( datetimes.length > 0 ) {
+			return datetimes.map( ( datetime ) => <li key={ datetime.id }>
+				{ datetime.start.toFormat() }
+				<br />
+				<CloneDateTimeButton
+					datetime={ datetime }
+					onCloneDateTime={ onDateClone }
+				/>
+			</li> );
+		}
+		return <li><Spinner /></li>;
+	}
+	return <div>
+		<h2>Datetimes List</h2>
+		<ul>{ getDatetimesList() }</ul>
+	</div>;
+};
+
+const DatetimesList = withDispatch( ( dispatch, ownProps, { select } ) => {
+	async function onDateClone( datetime ) {
+		const { getRelatedEntities } = select( 'eventespresso/core' );
+		const { createEntity, createRelations } = dispatch( 'eventespresso/core' );
+		const tickets = getRelatedEntities( datetime, 'tickets' );
+		const newDateTime = await createEntity( 'datetime', datetime.forClone );
+		createRelations(
+			'datetime',
+			newDateTime.id,
+			'tickets',
+			tickets
+		);
+		createRelations(
+			'event',
+			ownProps.eventId,
+			'datetimes',
+			[ newDateTime ]
+		);
+	}
+	return { onDateClone };
+} )( DatesList );
+
+const TicketsList = ( { tickets } ) => {
+	function getTicketsList() {
+		if ( tickets.length > 0 ) {
+			return tickets.map( ( ticket ) => <li key={ ticket.id }>{ ticket.name }</li> );
+		}
+		return <li><Spinner /></li>;
+	}
+	return <div>
+		<h2>Tickets List</h2>
+		<ul>{ getTicketsList() }</ul>
+	</div>;
+};
+
+const DateAndTicketList = ( {
+	eventId,
+	datetimes = [],
+	tickets = [],
+	isInitializing,
+} ) => {
+	function getContent() {
+		return isInitializing ?
+			<Spinner /> :
+			<Fragment>
+				<DatetimesList eventId={ eventId } datetimes={ datetimes } />
+				<TicketsList tickets={ tickets } />
+			</Fragment>;
+	}
+	return <div>{ getContent() }</div>;
+};
+
+const DatetimeAndTicketList = compose( [
+	withState( { isInitializing: true } ),
+	withSelect( ( select, ownProps ) => {
+		const { isInitializing } = ownProps;
+		const { getEntitiesForModel } = select( 'eventespresso/core' );
+		if ( isInitializing ) {
+			return ownProps;
+		}
+		return {
+			datetimes: getEntitiesForModel( 'datetime' ),
+			tickets: getEntitiesForModel( 'ticket' ),
+		};
+	} ),
```

Comment:
This `withSelect` HOC takes care of retrieving the datetimes and tickets used for the display but only when `isInitializing` is false.  That way it can confidently use the general selectors once the state is populated by the main query.  From that point on any changes to the datetimes or tickets state will immediately get reflected in the rendered component.